Source/Core/Resource.js

```diff
@@ -860,28 +916,38 @@ define([
         // 3. It's a blob URI
         // 4. It doesn't have request headers and we preferBlob is false
         if (!xhrBlobSupported || this.isDataUri || this.isBlobUri || (!this.hasHeaders && !preferBlob)) {
-            return fetchImage(this, true);
+            return fetchImage(this, flipY);
         }
 
         var blobPromise = this.fetchBlob();
         if (!defined(blobPromise)) {
             return;
         }
 
+        if (FeatureDetection.supportsCreateImageBitmap()) {
+            return blobPromise
+                .then(function(blob) {
+                    var deferred = when.defer();
+                    Resource._Implementations.createImageBitmapFromBlob(blob, deferred, flipY);
+                    return deferred;
```

Comment:
I don't understand why we need to pass a deferred to `createImageBitmapFromBlob`, it looks like that function is promise based anyway and should just return the promise it creates.

Reply:
That's only to make it work with the `Resource._Implementations.createImage` which already has a deferred given to it. Although I guess if it needs to resolve a given deferred it can do:

```
Resource._Implementations.createImageBitmapFromBlob()
  .then(deferred.resolve)
  .otherwise(deferred.reject)
```

?

Reply:
Yes, I would push the deferred usage up to where it needs it.